python/torch_mlir/__init__.py

```diff
@@ -2,19 +2,26 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 # Also available under a BSD-style license. See LICENSE.
-
+import warnings
 from typing import Optional, Sequence, Union, List, Dict, Tuple
 from enum import Enum
 
 import sys
 from io import StringIO
 
-from torch._functorch.compile_utils import strip_overloads
-import torch
+try:
+    from torch._functorch.compile_utils import strip_overloads
+    import torch
+except ImportError:
+    warnings.warn("PyTorch not installed")
 
 from torch_mlir.passmanager import PassManager
 from .compiler_utils import run_pipeline_with_repro_report
-from torch_mlir.dialects.torch.importer.jit_ir import ClassAnnotator, ImportOptions, ModuleBuilder
+from torch_mlir.dialects import torch as torch_dialect
```

Comment:
This artificial split is necessary so that import machinery loads the modules in the `dialects` directory (since it's not a module itself - no `__init__.py`) even if JIT IR importer isn't installed.